src/java/voldemort/store/metadata/MetadataStore.java

```diff
@@ -947,58 +968,28 @@ public void deleteRebalancingState(RebalanceTaskInfo stealInfo) {
         }
     }
 
-    /**
-     * change server state between OFFLINE_SERVER and NORMAL_SERVER
-     * 
-     * @param setToOffline True if set to OFFLINE_SERVER
-     */
-    public void setOfflineState(boolean setToOffline) {
+    public void setState(VoldemortState newState) {
         // acquire write lock
         writeLock.lock();
         try {
-            String currentState = ByteUtils.getString(get(SERVER_STATE_KEY, null).get(0).getValue(),
-                                                      "UTF-8");
-            if(setToOffline) {
-                // from NORMAL_SERVER to OFFLINE_SERVER
-                if(currentState.equals(VoldemortState.NORMAL_SERVER.toString())) {
-                    put(SERVER_STATE_KEY, VoldemortState.OFFLINE_SERVER);
-                    initCache(SERVER_STATE_KEY);
-                    put(SLOP_STREAMING_ENABLED_KEY, false);
-                    initCache(SLOP_STREAMING_ENABLED_KEY);
-                    put(PARTITION_STREAMING_ENABLED_KEY, false);
-                    initCache(PARTITION_STREAMING_ENABLED_KEY);
-                    put(READONLY_FETCH_ENABLED_KEY, false);
-                    initCache(READONLY_FETCH_ENABLED_KEY);
-                } else if(currentState.equals(VoldemortState.OFFLINE_SERVER.toString())) {
-                    logger.warn("Already in OFFLINE_SERVER state.");
-                    return;
-                } else {
-                    logger.error("Cannot enter OFFLINE_SERVER state from " + currentState);
```

Comment:
The new code does not attempt to prevent some transitions.  Is this a deliberate policy?